core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -91,7 +90,12 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
   /** List all the files currently stored on disk by the disk manager. */
   def getAllFiles(): Seq[File] = {
     // Get all the files inside the array of array of directories
-    subDirs.flatten.filter(_ != null).flatMap { dir =>
+    subDirs.flatMap { dir =>
```

Comment:
Well, you'd be claiming then that `File.mkdir()` may not actually create a directory (assuming no errors). That kind of update isn't 'thread-local' and can't be a question of the Java memory model.

I'm OK with merging this since it does indeed not even get called outside of tests.